planner/core/logical_plan_builder.go

```diff
@@ -596,13 +597,32 @@ func (b *PlanBuilder) buildProjectionField(id, position int, field *ast.SelectFi
 }
 
 // buildProjection returns a Projection plan and non-aux columns length.
-func (b *PlanBuilder) buildProjection(p LogicalPlan, fields []*ast.SelectField, mapper map[*ast.AggregateFuncExpr]int) (LogicalPlan, int, error) {
+func (b *PlanBuilder) buildProjection(p LogicalPlan, fields []*ast.SelectField, mapper map[*ast.AggregateFuncExpr]int, considerWindow bool) (LogicalPlan, int, error) {
 	b.optFlag |= flagEliminateProjection
 	b.curClause = fieldList
 	proj := LogicalProjection{Exprs: make([]expression.Expression, 0, len(fields))}.Init(b.ctx)
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(fields))...)
 	oldLen := 0
-	for _, field := range fields {
+	for i, field := range fields {
+		if !field.Auxiliary {
+			oldLen++
+		}
+
+		isWindowFuncField := ast.HasWindowFlag(field.Expr)
+		// When `considerWindow` is false, we will only build fields for non-window functions, so we add fake placeholders.
```

Comment:
add fake placeholders ... for window functions?

Reply:
In the execution phase, it will just be an empty column?

Reply:
Yes, it is for window functions, and the placeholders will be removed when do column pruning.

Reply:
What about adding more comments to help others understand these codes.

Reply:
Added.